js/snippetPreview.js

```diff
@@ -0,0 +1,162 @@
+ */
+YoastSEO_SnippetPreview.prototype.init = function() {
+    if( this.refObj.inputs.pageTitle !== null && this.refObj.inputs.url !== null ) {
+        this.output = this.htmlOutput();
+        this.renderOutput();
+    }
+};
+
+/**
+ * creates html object to contain the strings for the snippetpreview
+ * @returns {html object with html-strings}
+ */
+YoastSEO_SnippetPreview.prototype.htmlOutput = function() {
+    var html = {};
+    html.title = this.formatTitle();
+    html.cite = this.formatCite();
+    html.meta = this.formatMeta();
+    return html;
+};
+
+/**
+ * formats the title for the snippet preview
+ * @returns {formatted page title}
+ */
+YoastSEO_SnippetPreview.prototype.formatTitle = function() {
+    var title = this.refObj.inputs.pageTitle;
+    title = this.refObj.pageAnalyzer.YoastSEO_preProcessor.stripAllTags( title );
+    return this.formatKeyword( title );
+};
+
+/**
+ * formats the url for the snippet preview
+ * @returns formatted url
+ */
+YoastSEO_SnippetPreview.prototype.formatCite = function() {
+    var cite = this.refObj.inputs.url;
+    cite = this.refObj.pageAnalyzer.YoastSEO_preProcessor.stripAllTags( cite );
+    return this.formatKeywordUrl( cite );
+};
+
+/**
+ * formats the metatext for the snippet preview, if empty runs getMetaText
+ * @returns formatted metatext
+ */
+YoastSEO_SnippetPreview.prototype.formatMeta = function() {
+    var meta = this.refObj.inputs.meta;
+    if(meta === ""){
+        meta = this.getMetaText();
+    }
+    meta = this.refObj.pageAnalyzer.YoastSEO_preProcessor.stripAllTags( meta );
+    meta = meta.substring(0,analyzerConfig.maxMeta);
+    return this.formatKeyword( meta );
+};
+
+/**
+ * formats the metatext, based on the keyword to select a part of the text.
+ * If no keyword matches, takes the first 156chars (depending on the config)
+ * @returns metatext
+ */
+YoastSEO_SnippetPreview.prototype.getMetaText = function() {
+    var indexMatches = this.getIndexMatches();
+    var periodMatches = this.getPeriodMatches();
+    var metaText = this.refObj.inputs.textString.substring(0, analyzerConfig.maxMeta);
+    var curStart = 0;
+    if(indexMatches.length > 0) {
+        for (var j = 0; j < periodMatches.length; ) {
+            if (periodMatches[0] < indexMatches[0] ) {
+                curStart = periodMatches.shift();
+            } else {
+                if( curStart > 0 ){
+                    curStart += 2;
+                }
+                break;
+            }
+        }
+        metaText = this.refObj.inputs.textString.substring( curStart, curStart + analyzerConfig.maxMeta );
+    }
+    return metaText;
+};
+
+/**
+ * Builds an array with all indexes of the keyword
+ * @returns Array with matches
+ */
+YoastSEO_SnippetPreview.prototype.getIndexMatches = function() {
+    var indexMatches = [];
+    var match;
+    var i = 0;
+    while ( ( match = this.refObj.inputs.textString.indexOf( this.refObj.inputs.keyword, i ) ) > -1 ) {
```

Comment:
Perhaps define match on a new line and explain with inline comment what's happening as the while condition is getting unreadable